src/ComposerRequireChecker/Cli/CheckCommand.php

```diff
@@ -93,6 +95,14 @@ protected function execute(InputInterface $input, OutputInterface $output): int
                 )
             )
         );
+
+        if (
```

Comment:
Overall, the approach seems much more correct now, as it correctly appends `Composer\InstalledVersions` to the declared "found" symbols.

The high level behavior is correct, but we're still operating in the wrong code unit. 

I'm fine with merging this code as-is by adding a `@TODO`, but if you look above, the sources do come from:

```php
                    $getAdditionalSourceFiles($options->getScanFiles(), dirname($composerJson)),
                    $getPackageSourceFiles($composerData, dirname($composerJson)),
                    (new LocateComposerPackageDirectDependenciesSourceFiles())->__invoke($composerJson)
```

I think we need to define and call `$getComposerPluginApi($composerJson)` in that section (by adding a parameter and a dedicated class that is only responsible for this dependency check).

What's your opinion?

Reply:
I originally thought so too, but when I have passed path to InstalledVersions.php file to LocateDefinedSymbolsFromASTRoots, I found out with the fact that the file contains `use Composer\Semver\VersionParser` and `composer/semver` is not required in my project dependencies. As result I've got a message about unknown symbol `Composer\Semver\VersionParser`. Why composer does this — I don't know, but I don't think that everyone who use `composer-runtime-api` should require `composer/semver`.

Look at the last changes. Is it better solution?

Reply:
I made some refactoring and moved code that work with composerData to the dedicated locator class.

I can't use `LocateASTFromFiles` and `LocateDefinedSymbolsFromASTRoots` for get `Composer\InstalledVersions` from InstalledVersions.php because it contains `use Composer\Semver\VersionParser` and `composer/semver` may not be required in composer.json. I originally thought the same as you, but have got a message about undefined symbol `Composer\Semver\VersionParser`.

Also I see that you write about composer-**plugin**-api, but I work in this PR with composer-**runtime**-api. Should I change code and process the both packages?

Reply:
@Ocramius could you give me a feedback about the last comment?

Reply:
This works for me and is more than sufficient, thank you! :+1: